SLE 2024-05 Release Notes (#201)
ni/install-systemlink-enterprise: release-notes/2024-05/README.md

# SystemLink Enterprise release 2024-05 Release Notes

The 2024-05 release bundle for SystemLink Enterprise has been published to <https://downloads.artifacts.ni.com>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.

## New Features and Behavior changes

> - `serviceregistry 0.14.19`

Reviewer: I think it would make more sense to document permission changes for the changed service, not for the service registry, which just hosts the UI configuration.

Reviewer 2: Since I cannot find the users for Gopal Sabarinathan and Kaviarasu Selvarasu, who have added these documentation notes, I did start a Teams group chat with them, + JD, + Surya, + Usha.

Reviewer 2: @JD-Robertson , have replaced `serviceregistry 0.14.19` with `workorder 0.4.35`, per input from Gopal Sabarinathan in Teams

# SystemLink Enterprise release 2024-05 Release Notes

The 2024-05 release bundle for SystemLink Enterprise has been published to <https://downloads.artifacts.ni.com>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.

## New Features and Behavior changes

> - Navigate SystemLink Enterprise faster with the intuitively organized UI.

Reviewer 3: Please update to: "Navigate SystemLink Enterprise faster with the intuitively organized Home page and the navigation pane."

Reviewer 2: Updated @vpstoynova

# SystemLink Enterprise release 2024-05 Release Notes

The 2024-05 release bundle for SystemLink Enterprise has been published to <https://downloads.artifacts.ni.com>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.

## New Features and Behavior changes

- Navigate SystemLink Enterprise faster with the intuitively organized UI.
- Visualize data tables in a data space to gain insights into your time-series data.
- Analyze test results with Jupyter Notebooks.
- Annotate steps with keywords from a data space.

> - Save custom column configurations for the product specifications grid.

Reviewer 4: We need to add one more item. I'm also working with Vera to add these to her publication:

- Define and apply custom states to a system

Reviewer 2: I've added this new bullet

Reviewer 3: I've also added the bullet to the list in the User Manual. You can see it here: https://ni-staging.zoominsoftware.io/docs/en-US/bundle/systemlink-enterprise/page/systemlink-enterprise-new-features-and-changes.html

Reviewer 3: @CIakab-NI Here are the links to the related topics:
⁠https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/analyzing-test-results-with-jupyter-notebooks.html
⁠https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/visualizing-data-tables-in-a-data-space.html
⁠https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/editing-test-result-keywords-from-a-data-space.html
⁠https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/creating-a-test-plan.html

Reviewer 2: I did check your list and updated mine to be identical.

Thanks for the links @vpstoynova - I did add the links to Github (for some reasone 2x of the links are not highlighted although added to the Readme.md).

- `workorder 0.4.35`

>     - As the work orders and test plans are unlinked, the privileges are split between workorders and tesplans as follows:

Reviewer 4: ```suggestion
    - The privileges for work orders and test plans have been split into distinct categories.
```

        - Create work orders -  `workorder: Create`
        - Modify work orders -  `workorder: Update`
        - Delete work orders -  `workorder: Delete`
        - List and view work orders -  `workorder: Query`
        - Create test plans -  `testplan: Create`
        - Modify test plans -  `testplan: Update`
        - Delete test plans -  `testplan: Delete`
        - List and view test plans -  `testplan: Query`
        - Access work orders web application - `workorder:AccessApplication`

>         - Access test plans web application - `testplan:AccessApplication`

Reviewer 4: We don't need to list the specific underlying privileges; e.g. `work order: Create` etc. Just listing the user facing strings is enough.

Reviewer 2: Have kept only the User Facing Strings and removed the last part of each bullet

> - `userservice 0.14.9`

Reviewer 4: This doesn't need to be a major bullet. Promote the sub bullet below this to a major bullet.

Reviewer 2: I did promote the sub-bullet to the same level as the major bullet; if you feel like it, we can remove the name and version of the container and just leave the remaining text.

Reviewer 2: @prestwick , feel free to review the updates and let me know if you need other changes. 
Note that I've also added the notices files, after getting the OK from Fred, so there's nothing pending on my side unless you have further comments/input on the last edits..

>       - Data Maintainer role has been provided with all actions for test plans and query action for test plan template: `testplan: *`, `testplantemplate:Query`

Reviewer 4: ```suggestion
      - The Data Maintainer role has been updated to include all privileges for managing test plans and the privilege to query test plan templates
```

## Helm Chart Breaking Changes

- `webserver 0.21`
    - Removed unused and undocumented routes: `/api/keys`, `/api/keys/duplicate`, `/user/api-key`
- `systemlink 0.25.63` Top Level Helm Chart

>     - We upgraded our RabbitMQ dependency from 3.12.x to 3.13.x. Minor updates to the RabbitMQ version are potentially breaking. We document this in our release notes because customers will need to install a version of the product running version 3.12 before they can successfully upgrade to a version running 3.13.

Reviewer 4: ```suggestion
    - We upgraded our RabbitMQ dependency from 3.12.x to 3.13.x. Minor updates to the RabbitMQ version are potentially breaking. Customers must install a version of SystemLink Enterprise running RabbitMQ 3.12 before they can successfully upgrade to a version running 3.13.
```

- `systemlink 0.25.63` Top Level Helm Chart

>     - Enabled the Work Order and Lab Management services by default. They are still an optional installation. To disable installation, set the enabled flag to false in the Work Order service configuration.

Reviewer 4: ```suggestion
    - The Work Order and Lab Management services are installed by default. You may disable installation for these services by setting the enabled flag to `false` in the Work Order service configuration.
```
